File: src/types/game.ts
export type GameCapability =
  | 'lan'
  | 'ip_join'
  | 'dedicated_server'
  | 'steam_lobby'
  | 'steam_p2p'
  | 'official_server'
  | 'unknown';

export type MultiplayerCapability =
  | 'native_lan_ip'
  | 'hidden_dedicated_server'
  | 'lan_discovery_broadcast'
  | 'tcp_udp_proxy_possible'
  | 'community_mod'
  | 'official_only'
  | 'unsupported'
  | 'unknown';

export type ConversionMethod =
  | 'virtual_lan'
  | 'dedicated_server_launcher'
  | 'broadcast_bridge'
  | 'port_proxy'
  | 'mod_installer'
  | 'steam_relay_plugin'
  | 'manual_guide'
  | 'not_supported';

export type GameNetworkType =
  | 'lan_ip_direct'
  | 'dedicated_server'
  | 'tcp_port_proxy_needed'
  | 'udp_broadcast_needed'
  | 'steam_lobby_direct_possible'
  | 'steam_relay_plugin'
  | 'mod_required'
  | 'official_only'
  | 'not_supported'
  | 'unknown_need_review';

export interface GameConnectionPlan {
  summary: string;
  host_role: string;
  join_role: string;
  default_join_host?: string | null;
  default_join_port?: number | null;
  requires_virtual_lan: boolean;
  requires_tcp_port_proxy: boolean;
  requires_udp_broadcast_bridge: boolean;
  requires_dedicated_server: boolean;
  invite_template: string[];
  troubleshooting: string[];
}

export interface MultiplayerConversionProfile {
  capability: MultiplayerCapability;
  methods: ConversionMethod[];
  can_convert_to_lan: boolean;
  risk_level: 'low' | 'medium' | 'high';
  notes: string[];
  required_components: string[];
}

export interface LaunchConfigField {
  id: string;
  label: string;
  type: 'text' | 'number' | 'password' | 'select' | 'checkbox';
  default_value?: string | null;
  required?: boolean | null;
  help?: string | null;
  options?: string[] | null;
}

export interface LaunchProfile {
  id: string;
  name: string;
  type: 'client' | 'server' | 'docs';
  exe?: string | null;
  args?: string[] | null;
  arg_templates?: string[] | null;
  stdin_templates?: string[] | null;
  config_fields?: LaunchConfigField[] | null;
}

export interface GameSummary {
  game_id: string;
  display_name: string;
  steam_appid?: string | null;
  detected_path?: string | null;
  capabilities: GameCapability[];
  multiplayer_conversion?: MultiplayerConversionProfile | null;
  network_type?: GameNetworkType;
  connection_plan?: GameConnectionPlan;
  adapter_source?: 'builtin' | 'registry' | 'custom' | 'steam_scan' | string;
}

export interface GameAnalysis extends GameSummary {
  confidence: 'high' | 'medium' | 'low';
  notes: string[];
  launch_profiles: LaunchProfile[];
  default_ports: number[];
}

export interface GameAdapter {
  game_id: string;
  display_name: string;
  steam_appid?: string | null;
  capabilities: GameCapability[];
  multiplayer_conversion?: MultiplayerConversionProfile | null;
  network_type?: GameNetworkType;
  connection_plan?: GameConnectionPlan;
  adapter_source?: 'builtin' | 'registry' | 'custom' | 'steam_scan' | string;
  executables: string[];
  default_ports: number[];
  launch_profiles: LaunchProfile[];
}

File: src-tauri/src/models/game.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum GameCapability {
    Lan,
    IpJoin,
    DedicatedServer,
    SteamLobby,
    SteamP2p,
    LocalCoop,
    RemotePlayTogether,
    OfficialServer,
    Unknown,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum MultiplayerCapability {
    NativeLanIp,
    HiddenDedicatedServer,
    LanDiscoveryBroadcast,
    TcpUdpProxyPossible,
    LocalCoopRemotePlay,
    SteamP2pLobby,
    CommunityMod,
    OfficialOnly,
    Unsupported,
    Unknown,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum ConversionMethod {
    VirtualLan,
    DedicatedServerLauncher,
    BroadcastBridge,
    PortProxy,
    ModInstaller,
    SteamRelayPlugin,
    SteamRemotePlay,
    SunshineMoonlight,
    WireguardGuide,
    ZerotierGuide,
    TailscaleGuide,
    ManualGuide,
    NotSupported,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum GameNetworkType {
    LanIpDirect,
    DedicatedServer,
    TcpPortProxyNeeded,
    UdpBroadcastNeeded,
    SteamLobbyDirectPossible,
    SteamRelayPlugin,
    LocalCoopRemotePlay,
    SteamP2pOnly,
    ModRequired,
    OfficialOnly,
    NotSupported,
    UnknownNeedReview,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GameConnectionPlan {
    pub summary: String,
    pub host_role: String,
    pub join_role: String,
    pub default_join_host: Option<String>,
    pub default_join_port: Option<u16>,
    pub requires_virtual_lan: bool,
    pub requires_tcp_port_proxy: bool,
    pub requires_udp_broadcast_bridge: bool,
    pub requires_dedicated_server: bool,
    pub invite_template: Vec<String>,
    pub troubleshooting: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MultiplayerConversionProfile {
    pub capability: MultiplayerCapability,
    pub methods: Vec<ConversionMethod>,
    pub can_convert_to_lan: bool,
    pub risk_level: String,
    pub notes: Vec<String>,
    pub required_components: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AdapterApplicabilityProfile {
    pub verification_status: String,
    pub tested_versions: Vec<String>,
    pub tested_platforms: Vec<String>,
    pub supported_os: Vec<String>,
    pub network_conditions: Vec<String>,
    pub known_limitations: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AdapterEvidenceProfile {
    pub port_protocols: Vec<String>,
    pub proof_items: Vec<String>,
    pub test_steps: Vec<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub last_verified_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LaunchConfigField {
    pub id: String,
    pub label: String,
    #[serde(rename = "type")]
    pub field_type: String,
    pub default_value: Option<String>,
    pub required: Option<bool>,
    pub help: Option<String>,
    pub options: Option<Vec<String>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LaunchProfile {
    pub id: String,
    pub name: String,
    #[serde(rename = "type")]
    pub profile_type: String,
    pub exe: Option<String>,
    pub args: Option<Vec<String>>,
    pub arg_templates: Option<Vec<String>>,
    pub stdin_templates: Option<Vec<String>>,
    pub config_fields: Option<Vec<LaunchConfigField>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GameSummary {
    pub game_id: String,
    pub display_name: String,
    pub steam_appid: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub adapter_version: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
    pub detected_path: Option<String>,
    pub capabilities: Vec<GameCapability>,
    pub multiplayer_conversion: Option<MultiplayerConversionProfile>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub network_type: Option<GameNetworkType>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub connection_plan: Option<GameConnectionPlan>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub applicability: Option<AdapterApplicabilityProfile>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub evidence: Option<AdapterEvidenceProfile>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub adapter_source: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GameAnalysis {
    pub game_id: String,
    pub display_name: String,
    pub steam_appid: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub adapter_version: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
    pub detected_path: Option<String>,
    pub capabilities: Vec<GameCapability>,
    pub multiplayer_conversion: Option<MultiplayerConversionProfile>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub network_type: Option<GameNetworkType>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub connection_plan: Option<GameConnectionPlan>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub applicability: Option<AdapterApplicabilityProfile>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub evidence: Option<AdapterEvidenceProfile>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub adapter_source: Option<String>,
    pub confidence: String,
    pub notes: Vec<String>,
    pub launch_profiles: Vec<LaunchProfile>,
    pub default_ports: Vec<u16>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GameAdapter {
    pub game_id: String,
    pub display_name: String,
    pub steam_appid: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub adapter_version: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
    pub capabilities: Vec<GameCapability>,
    pub multiplayer_conversion: Option<MultiplayerConversionProfile>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub network_type: Option<GameNetworkType>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub connection_plan: Option<GameConnectionPlan>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub applicability: Option<AdapterApplicabilityProfile>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub evidence: Option<AdapterEvidenceProfile>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub adapter_source: Option<String>,
    pub executables: Vec<String>,
    pub default_ports: Vec<u16>,
    pub launch_profiles: Vec<LaunchProfile>,
}
